@here/harp-mapview/lib/Utils.ts

```diff
@@ -201,6 +193,59 @@ export namespace MapViewUtils {
         );
     }
 
+    /**
+     * Calculate target (focus) point geo-coordinates for given camera.
+     * @see getTargetPositionFromCamera
+     *
+     * @param camera The camera looking on target point.
+     * @param projection The geo-projection used.
+     * @param elevation Optional elevation above (or below) sea level measured in world units.
+     */
+    export function getTargetCoordinatesFromCamera(
+        camera: THREE.Camera,
+        projection: Projection,
+        elevation?: number
+    ): GeoCoordinates | null {
+        // This function does almost the same as:
+        // rayCastGeoCoordinates(mapView, 0, 0)
+        // but in more gentle and performance wise manner
+        const targetWorldPos = getTargetPositionFromCamera(camera, projection, elevation);
+        if (targetWorldPos !== null) {
+            return projection.unprojectPoint(targetWorldPos);
+        }
+        return null;
+    }
+
+    /**
+     * Calculate target (focus) point world coordinates for given camera position and orientation.
+     * @param camera The camera looking on target point.
+     * @param projection The geo-projection used.
+     * @param elevation Optional elevation above (or below) sea level in world units.
+     */
+    export function getTargetPositionFromCamera(
+        camera: THREE.Camera,
+        projection: Projection,
+        elevation?: number
+    ): THREE.Vector3 | null {
+        const cameraPos = cache.vector3[0].copy(camera.position);
+        const cameraLookAt = camera.getWorldDirection(cache.vector3[1]);
+        rayCaster.set(cameraPos, cameraLookAt);
+        if (elevation !== undefined) {
+            groundPlane.constant = -elevation;
+            groundSphere.radius += elevation;
+        }
+        const targetWorldPos = new THREE.Vector3();
+        const result =
+            projection.type === ProjectionType.Planar
+                ? rayCaster.ray.intersectPlane(groundPlane, targetWorldPos)
+                : rayCaster.ray.intersectSphere(groundSphere, targetWorldPos);
+        if (elevation !== undefined) {
+            groundPlane.constant = 0;
+            groundSphere.radius -= elevation;
```

Comment:
Maybe a bit paranoid, but due to floating point precision resetting it to the original value would be better.

Reply:
Done.